EdgeDeflector/resources/nsis_installer.nsi

```diff
@@ -1,72 +1,237 @@
 ; Installer for EdgeDeflector
 
+
+;--------------------------------
+;Include Modern UI
+
+  !include "MUI2.nsh"
+
 ;--------------------------------
+;General
+
+  ;Define name of the product
+  !define PRODUCT "EdgeDeflector"
+  !define ABOUTURL "https://github.com/da2x/EdgeDeflector"
+  !define INSTALLATIONURL "https://github.com/da2x/EdgeDeflector/wiki/Thank-you-for-installing-EdgeDeflector"
+
+  ;Properly display all languages
+  Unicode true
+
+  ;Show 'console' in installer and uninstaller
+  ShowInstDetails "show"
+  ShowUninstDetails "show"
+
+  ;Name and file
+  Name "${PRODUCT}"
+  Icon "..\Icon.ico"
+  LicenseData "..\..\LICENSE"
+  OutFile "..\bin\Release\${PRODUCT}_install.exe"
 
-!define PRODUCT "EdgeDeflector"
+  ;Version
+  !define VERSIONMAJOR 1
+  !define VERSIONMINOR 1
+  !define VERSIONBUILD 1
 
-Name "${PRODUCT} Installer"
+  ;Default installation folder
+  InstallDir "$PROGRAMFILES\${PRODUCT}"
 
-OutFile "..\bin\Release\${PRODUCT}_install.exe"
+  ;Get installation folder from registry if available
+  InstallDirRegKey HKLM "Software\${PRODUCT}" ""
 
-; Default installation directory
-InstallDir $PROGRAMFILES\${PRODUCT}
+  ;Request application admin
+  RequestExecutionLevel admin
 
-; Store install dir in the registry
-InstallDirRegKey HKLM "Software\${PRODUCT}" "Install_Dir"
+;--------------------------------
+;Interface Settings
+
+  ;Show warning if user wants to abort
+  !define MUI_ABORTWARNING
 
-; Request application privileges for UAC
-RequestExecutionLevel admin
+  ;Show all languages, despite user's codepage
+  !define MUI_LANGDLL_ALLLANGUAGES
 
+  ;Use the custom own icon
+  !define MUI_ICON "..\Icon.ico"
+  !define MUI_UNICON "UnIcon.ico"
+  !define MUI_HEADERIMAGE_RIGHT
+  !define MUI_WELCOMEFINISHPAGE_BITMAP "installer_right.bmp"
+  !define MUI_UNWELCOMEFINISHPAGE_BITMAP "uninstaller_right.bmp"
 
 ;--------------------------------
+;Language Selection Dialog Settings
 
-; Installer pages
-Page directory
-Page instfiles
+  ;Remember the installer language
+  !define MUI_LANGDLL_REGISTRY_ROOT "HKCU" 
+  !define MUI_LANGDLL_REGISTRY_KEY "Software\${PRODUCT}" 
+  !define MUI_LANGDLL_REGISTRY_VALUENAME "Installer Language"
+
+;--------------------------------
+;Pages
+
+  ;For the installer
+  !insertmacro MUI_PAGE_WELCOME
+  !insertmacro MUI_PAGE_LICENSE "..\..\LICENSE"
+  #!insertmacro MUI_PAGE_COMPONENTS
```

Comment:
Why is this commented out?

Reply:
My bad - in my special fork this was used - it can be deleted.

Reply:
My bad - in my fork this was used - it can be deleted now.